cmdline/haskell-gi.hs

```diff
@@ -71,6 +71,20 @@ parseKeyValue s =
   let (a, '=':b) = break (=='=') s
    in (a, b)
 
+solveNameVersion::Overrides -> Text -> (Text, Maybe Text)
+solveNameVersion ovs name = (namewithoutversion, version)
+ where
+   namepieces = T.splitOn "-" name
+   namewithoutversion = head namepieces
+   versioninname = if length namepieces == 2
+                     then Just (last namepieces)
+                     else Nothing
+   versioninoveride = M.lookup namewithoutversion (nsChooseVersion ovs)
+   version = if isNothing versioninoveride
```

Comment:
versionInOverride and versionInName type is Maybef
romMaybe not fit here, since there is no default value. any suggestion?